src/inmanta/user_setup.py

```diff
@@ -0,0 +1,163 @@
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import asyncio
+import secrets
+import socket
+
+import asyncpg
+import click
+
+import nacl.pwhash
+from inmanta import config, data
+from inmanta.data import CORE_SCHEMA_NAME, PACKAGE_WITH_UPDATE_FILES, schema
+from inmanta.server import config as server_config
+
+
+def generate_signing_config() -> str:
+    hostname = socket.gethostname()
+    return f"""[auth_jwt_default]
+algorithm=HS256
+sign=true
+client_types=agent,compiler,api
+key={secrets.token_urlsafe(32)}
+expire=0
+issuer=https://{hostname}:{server_config.server_bind_port.get()}/
+audience=https://{hostname}:{server_config.server_bind_port.get()}/
+"""
+
+
+def validate_server_setup() -> None:
+    """Validate the server configuration so that authentication is setup correctly."""
+    config.Config.load_config()
+
+    # make sure auth is on
+    if not server_config.server_enable_auth.get():
+        raise click.ClickException(
+            "Server authentication should be enabled before running the initial user setup. "
+            "The option auth in the server section should be enabled."
+        )
+
+    click.echo(f"{'Server authentication: ' : <50}{click.style('enabled', fg='green')}")
+
+    # make sure the method is set to database
+    if server_config.server_auth_method.get() != "database":
+        raise click.ClickException(
+            "The server authentication method should be set to database to continue. Make sure auth_method in the server "
+            "section is set to database"
+        )
+
+    click.echo(f"{'Server authentication method: ' : <50}{click.style('database', fg='green')}")
+
+    # make sure there is auth config that supports signing tokens
+    cfg = config.AuthJWTConfig.get_sign_config()
+    if cfg is None:
+        click.echo("Error: No signing config available in the configuration.")
+
+        value = None
+        while value not in ["yes", "no"]:
+            value = click.prompt("Do you want to generate a new configuration? yes/no")
+
+        if value == "yes":
+            click.echo("Add the following to the configuration in /etc/inmanta/inmanta.d/auth.cfg:\n")
+            click.echo(generate_signing_config())
+
+        raise click.ClickException("Make sure signing configuration is added to the config. See the documentation for details.")
+
+    click.echo(f"{'Authentication signing config: ' : <50}{click.style('enabled', fg='green')}")
```

Comment:
`enabled` is not really the right word here for a check that verifies the presence of a config. I am not sure whether `found` is the perfect word here, but it's definitely more accurate than enabled.

```suggestion
    click.echo(f"{'Authentication signing config: ' : <50}{click.style('found', fg='green')}")
```